### PR TITLE
Image in the product title is not properly displayed, display dummy image if no image available, roduct title, code, name, und decsription should be truncated

### LINKED ISSUE #11551: Image in the product title is not properly displayed, product title, name, code and description should be truncated
image
Here there is a number of **breaking changes**:
1. `cx-media` component should be added to the HTML and img element should be removed,
2. `MediaModule` should be added as import to `*.module.ts` file,
2. 2 `public` method should be removed: `getProductImageURL` and `getProductImageAlt`
3. `*.spect.ts` file should be updated accordingly.

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

**Comment by Uli-Tiger** [comment]
@Larisa-Staroverova : 9 Unit tests are failing, probably needs a merge from with epic/cpq-configurator

File: feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts [hunk 1]
```diff
@@ -48,10 +48,16 @@ export class ConfiguratorProductTitleComponent {
     this.showMore = !this.showMore;
   }
 
+  /**
+   * @deprecated since 3.1
+   */
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts:53]: can you add a todo here or change it to 3.3 (currently we plan to release with 3.3)

File: feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.spec.ts [hunk 2]
```diff
@@ -155,6 +162,12 @@ describe('ConfigProductTitleComponent', () => {
             useClass: MockProductService,
           },
           { provide: IconLoaderService, useClass: MockIconFontLoaderService },
+          {
+            provide: FeaturesConfig,
+            useValue: {
+              features: { level: '3.1' },
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.spec.ts:168]: can we already 3.3 here? (most likely we release with 3.3)

File: feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss [hunk 3]
```diff
@@ -77,12 +88,26 @@
           // A larger value than the details need, this will be used to animate the closing/opening.
           // If the value is to large the closing will "delay" as the max-height will be reduced from the specified value down to 0.
           max-block-size: 400px;
+
+          @include forVersion(3.1) {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss:92]: change to 3.3 or add an todo

File: feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss [hunk 4]
```diff
@@ -77,12 +88,26 @@
...
+            }
+          }
         }
 
         .cx-details-image {
           margin-block-start: 15px;
           margin-block-end: 0px;
           max-inline-size: 100px;
+
+          @include forVersion(3.1) {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss:108]: change to 3.3 or add an todo

File: feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss [hunk 5]
```diff
@@ -101,6 +130,16 @@
           .cx-code {
             color: var(--cx-color-secondary);
           }
+
+          .cx-detail-title,
+          .cx-code,
+          .cx-description {
+            @include forVersion(3.1) {
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss:137]: change to 3.3 or add an todo

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]
looks good - But as far as i know, we plan to release with 3.3, so we should just use version 3.3 everywhere instead of 3.1

File: feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts:53]: Done!

File: feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.spec.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.spec.ts:168]: Done!

File: feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss (hunk 3 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss:92]: The version will be updated accordingly, see #11238

File: feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss:108]: The version will be updated accordingly, see #11238

File: feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss (hunk 5 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/styles/_configurator-product-title.scss:137]: The version will be updated accordingly, see #11238

**Review by Uli-Tiger — APPROVED** [review]

**Comment by Uli-Tiger** [comment]
unit test passed: https://travis-ci.com/github/SAP/spartacus/jobs/495452462

**Pull request merged**